src/PostgREST/QueryBuilder.hs

```diff
@@ -173,10 +173,7 @@ readRequestToCountQuery (Node (Select{from=qi, where_=logicForest}, _) _) =
 limitedQuery :: H.Snippet -> Maybe Integer -> H.Snippet
 limitedQuery query maxRows = query <> H.sql (maybe mempty (\x -> " LIMIT " <> BS.pack (show x)) maxRows)
 
-setLocalQuery :: Text -> (Text, Text) -> SqlQuery
-setLocalQuery prefix (k, v) =
-  "SET LOCAL " <> pgFmtIdent (prefix <> k) <> " = " <> pgFmtLit v <> ";"
-
-setLocalSearchPathQuery :: [Text] -> SqlQuery
-setLocalSearchPathQuery vals =
-  "SET LOCAL search_path = " <> BS.intercalate ", " (pgFmtLit <$> vals) <> ";"
+-- | Do a pg set_config(setting, value, true) call. This is equivalent to a SET LOCAL.
+setConfigLocal :: Text -> (Text, Text) -> H.Snippet
+setConfigLocal prefix (k, v) =
+  "set_config(" <> H.sql (pgFmtLit (prefix <> k)) <> ", " <> unknownLiteral v <> ", true)"
```

Comment:
Do I read this correctly as: The first argument is not parametrized, but the value is?

The first argument (setting) needs to be parametrized as well, because in some cases this is user input: Header names, including custom ones, are sent by the client.

Reply:
Yeah. I thought that the `set_config('search_path', val, true)` and `set_config('role', val, true)` should be really visible in the logs. I didn't want to parametrize them because they're set by us too.

I guess I'll just parametrize all now.